### DISCUSSION TITLE
WatchYourLAN Python client and HomeAssistant integration

### DISCUSSION THREAD

**Discussion by drwahl** [aceberg/WatchYourLAN#121] [Show and tell]
I created a python client library to access the WatchYourLAN from python:
https://github.com/drwahl/py-watchyourlanclient
https://pypi.org/project/watchyourlanclient/

I also have a couple of PRs up to get WatchYourLAN added to HomeAssistant as an official integration:
https://github.com/home-assistant/home-assistant.io/pull/34584
https://github.com/home-assistant/core/pull/125238

I wanted to share these and hopefully get some sort of "official" permission to support these features. I'm happy to maintain the library and HomeAssistant integration, but it didn't feel right doing so without some sort of permission.

**Comment by aceberg** [comment]
That's great! I'll add these links to the documentation a bit later.
And if you want to maintain a HomeAssistant integration, I'm absolutely ok with it. Please, send me a link and it will be added to the docs too.

**Comment by BenJamesAndo** [comment]
I'm looking forward to the HA integration.
I saw someones made a WatchYourLAN Addon for HA https://github.com/jeankhawand/addon-watchyourlan

**Comment by drwahl** [comment]
I'm going through the process of getting this upstreamed with HomeAssistant. It's not the quickest process, but it should result in some good code. I'll let you know once this has been accepted upstream and I'll include the official links to everything once they are in place.

**Comment by aceberg** [comment]
Added a link to Python API client in the README in `API & Integrations` section.

**Comment by drwahl** [comment]
Just wanted to drop another update: I'm still actively going through the code review process with the HA team, but I haven't abandoned this yet.

**Reply by J0han3s** [reply]
hi,
Can i ask if this got anywhere please?

Thanks

**Comment by jzucker2** [comment]
Have you thought about making a HACS integration first?

Like: https://www.hacs.xyz/

I have made a few. Pretty simple and it allows you to use it before getting the home assistant integration.
